src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random randomGenerator = new Random();
+    private static final ColorSupplier COLOR_SUPPLIER = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (randomGenerator.nextInt(5)) {

```

Comment:
The number `5` used in `randomGenerator.nextInt(5)` is a magic number. Consider defining it as a constant, such as `FIGURE_COUNT`, to improve code readability and maintainability.